internal/dataplane/parser/translate_tcproute.go

```diff
@@ -38,39 +38,27 @@ func (p *Parser) ingressRulesFromTCPRoutes() ingressRules {
 		applyExpressionToIngressRules(&result)
 	}
 
-	if len(errs) > 0 {
-		for _, err := range errs {
-			p.logger.Error(err, "could not generate route from TCPRoute")
-		}
+	for _, err := range errs {
+		p.logger.Error(err, "could not generate route from TCPRoute")
 	}
 
 	return result
 }
 
 func (p *Parser) ingressRulesFromTCPRoute(result *ingressRules, tcproute *gatewayapi.TCPRoute) error {
-	// first we grab the spec and gather some metdata about the object
 	spec := tcproute.Spec
-
-	// validation for TCPRoutes will happen at a higher layer, but in spite of that we run
-	// validation at this level as well as a fallback so that if routes are posted which
-	// are invalid somehow make it past validation (e.g. the webhook is not enabled) we can
-	// at least try to provide a helpful message about the situation in the manager logs.
```

Comment:
I believe this  comment was beneficial to an extent. It was obvious why we need it (as a fallback strategy if the admission webhook is disabled). Are we sure we want to remove it? 🤔 I'd say keep it but rephrase it (mentioning here the "higher layer", and what's that specifically. WDYT?